org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java

```diff
@@ -1234,5 +1235,45 @@ public void checkOppositeReferenceUsed(Assignment assignment) {
 			}
 		}
 	}
-
+	
+	@Check
+	public void checkAnnotations(AbstractRule rule) {
```

Comment:
How about an error validation that can be disabled in the preferences (or even using a system property)? Because some users would certainly be willing to add the declarations once that is required. Then they could at least already benefit from the new syntax. For many use cases (like @SuppressWarnings) that would of course also require some syntax for parameters.